### PR TITLE
Agent caching docs

### PR DESCRIPTION
**Pull request by calvn**

### CODE REVIEW

File: website/source/docs/agent/caching/index.html.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+sidebar_title: "Caching"
+sidebar_current: "docs-agent-caching"
+description: |-
+  Vault Agent's Caching functionality allows side-client caching of tokens and
+  secrets.
+---
+
+# Vault Agent Caching
+
+Vault Agent's Caching functionality allows side-client caching of tokens and
```
Comment (yhyakuna) [inline website/source/docs/agent/caching/index.html.md]: `side-client` ---> `client-side` maybe?

Comment (calvn) [reply website/source/docs/agent/caching/index.html.md]: whoops, that's what I meant!

File: website/source/docs/agent/caching/index.html.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  for the certificate.
+
+- `tls_cert_file` `(string: optional)` - Specifies the path to the certificate
+  for TLS.
+
+### Example Configuration
+
+An example configuration, with very contrived values, follows:
+
+```python
```
Comment (calvn) [inline website/source/docs/agent/caching/index.html.md]: python?

**Comment by calvn** [comment]
@jescalan how can we add this new page to the sidebar section?

File: website/data/docs_detailed_categories.yml [hunk 3]
```diff
@@ -65,6 +65,7 @@
   docs:
     - docs/agent/index.html
     - docs/agent/autoauth/index.html
+    - docs/agent/caching/index.html
```
Comment (kalafut) [inline website/data/docs_detailed_categories.yml]: Will there be a sidebar link too?

File: website/source/docs/agent/caching/index.html.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+1. Token creation requests are made through the agent and not directly to the
+   Vault server. Login endpoints from various auth methods also fall under this
+   category along with token creation endpoints of the token auth method.
+
+2. Leased secret creation requests are made through the agent using tokens that
+   are already managed by the agent.
+
+## Using Auto-Auth Token
+
+The requests that reach the agent, if they don't already bear a token as part
```
Comment (kalafut) [inline website/source/docs/agent/caching/index.html.md]: Slightly reworded: "If the requests that reach the agent don't already bear a token as part of the request, the auto-auth token can be put to use."

File: website/source/docs/agent/caching/index.html.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+renewal results in an error.
+
+Agent also does some best-effort cache evictions by observing specific request
+types and response codes. For example, if a token revocation request is made
+via the agent and if the request succeeds, then agent evicts all the cache
+entries associated with the revoked token. Similar behavior is exercised for
+lease revocations as well.
+
+While agent tries to observe some requests and evicts cache entries
+automatically, it is possible that agent is completely unaware of revocations
```
Comment (kalafut) [inline website/source/docs/agent/caching/index.html.md]: Isn't the agent completely unaware of revocations outside of the agent's context?  If so, remove "it is possible that".

File: website/source/docs/agent/caching/index.html.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+`/v1/agent/cache-clear`(see below) is made available to manually evict cache
+entries based on some of the criteria.
+
+## API
+
+### Cache Clear
+
+This endpoint clears the cache based on given parameters. To be able to use
+this API, some information on how the agent caches values should be known
+beforehand. Each response that gets cached in the agent are indexed on some
```
Comment (kalafut) [inline website/source/docs/agent/caching/index.html.md]:
```suggestion
beforehand. Each response that gets cached in the agent is indexed on some
```

File: website/data/docs_detailed_categories.yml (hunk 3 above)
Comment (calvn) [reply website/data/docs_detailed_categories.yml]: Do you know how to add it to the sidebar?

Comment (jescalan) [reply website/data/docs_detailed_categories.yml]: https://github.com/hashicorp/vault/tree/master/website#updating-navigation

**Comment by jescalan** [comment]
@calvn should be [in the readme](https://github.com/hashicorp/vault/tree/master/website#updating-navigation)!  Let me know if you're having trouble.

**Pull request merged**